avocado/utils/memory.py

```diff
@@ -320,3 +320,56 @@ def get_thp_value(feature):
         return (re.search(r"\[(\w+)\]", value)).group(1)
     else:
         return value
+
+
+class MemInfo(object):
+    """
+    Representation of /proc/meminfo
+    """
+
+    class MemInfoItem(object):
```

Comment:
First, defining classes inside of classes is unnecessary, unless there's a very compelling reason to, since it makes the code less readable. There's no such reason as I see it, apart from the fact that maybe you don't want to make the class directly acessible to other users, in which case you can simply prefix the class name with `_`.

Second, the MemInfoItem is really comprised by a label and a data size, so indeed it's not strictly a data size, but that comment is a bit on the nitpicky side. What I meant is that you could slightly refactor the code to create a DataSize class that could be useful for other things, then a MemInfoItem that uses DataSize. Sorry for not being clear earlier.

So there you have it, the problems I was exactly trying to solve with my request.

The suggestion of creating the 'pure' size related class stands, but there's always an argument against overengineering, that is, create something that is potentially useful but ends up not being useful, so I understand not wanting to do it. I'm fine with MemInfoItem as is.

The nested class argument is the one thing I would ask you to consider more seriously.